### PR TITLE
Add hash_combine

### PR DESCRIPTION
**Pull request by ShenMian**
hash_combine 实现来源: https://www.boost.org/doc/libs/1_62_0/doc/html/hash.html

### CODE REVIEW

**Comment by ShenMian** [comment]
看上去和 8961262e77a663e30be52daebc9410247f2d8eba 存在一样的构建错误, 等主分支修复完问题再重新运行 Github Actions 试试.

File: engine/source/runtime/function/physics/collision_detection.h [hunk 1]
```diff
@@ -32,22 +33,16 @@ namespace Pilot
...
-        bool operator==(const CollisionInfo& other_info) const
-        {
-            if (other_info.m_id_a == m_id_a && other_info.m_id_b == m_id_b)
-            {
-                return true;
-            }
-
-            return false;
-        }
+        bool operator==(const CollisionInfo& rhs) const { return rhs.m_id_a == m_id_a && rhs.m_id_b == m_id_b; }
```
Comment (Ol6rin) [inline engine/source/runtime/function/physics/collision_detection.h:45]: The modifications are good, but this file belongs to the deprecated legacy physics system, which will be deleted in the future.

File: engine/source/runtime/function/render/render_type.h [hunk 2]
```diff
@@ -1,5 +1,6 @@
 #pragma once
 
+#include "runtime/core/base/hash.h"
 #include <cstdint>
```
Comment (Ol6rin) [inline engine/source/runtime/function/render/render_type.h:4]: Add a blank line here to separate headers in the project and standard headers.

File: engine/source/runtime/function/render/render_type.h [hunk 3]
```diff
@@ -120,14 +121,13 @@ namespace Pilot
...
         {
-            size_t h0 = std::hash<std::string> {}(m_base_color_file);
-            size_t h1 = std::hash<std::string> {}(m_metallic_roughness_file);
-            size_t h2 = std::hash<std::string> {}(m_normal_file);
-            size_t h3 = std::hash<std::string> {}(m_occlusion_file);
-            size_t h4 = std::hash<std::string> {}(m_emissive_file);
-            return (((h0 ^ (h1 << 1)) ^ (h2 << 1)) ^ (h3 << 1)) ^ (h4 << 1);
+            size_t hash = 0;
+            hash_combine(
+                hash, m_base_color_file, m_metallic_roughness_file, m_normal_file, m_occlusion_file, m_emissive_file);
```
Comment (Ol6rin) [inline engine/source/runtime/function/render/render_type.h]: It seems that the formatting configure is wrong here, I'll submit a manual formatted version.

**Review by Ol6rin — CHANGES_REQUESTED** [review]

**Review by Ol6rin — APPROVED** [review]

**Pull request closed without merging**